torchbenchmark/score/compute_score.py

```diff
@@ -7,32 +7,130 @@
 import math
 import sys
 import os
+import re
 import yaml
+import importlib
 
 from tabulate import tabulate
+from pathlib import Path
+from collections import defaultdict
 
-SPEC_FILE_DEFAULT = "score.yml"
+SPEC_FILE_DEFAULT = "torchbenchmark/score/score.yml"
 TARGET_SCORE_DEFAULT = 1000
+TORCHBENCH_V0_SCORE = "torchbenchmark/score/torchbench_0.0.yaml"
+TORCHBENCH_V1_SCORE = "" # Placeholder for v1 score
 
-def compute_score(config, data, fake_data=None):
-    target = config['target']
-    score = 0.0
-    weight_sum = 0.0
-    for name in config['benchmarks']:
-        cfg = config['benchmarks'][name]
-        weight, norm = cfg['weight'], cfg['norm']
-        weight_sum += weight
-        measured_mean = [b['stats']['mean'] for b in data['benchmarks'] if b['name'] == name]
-        assert len(measured_mean) == 1, f"Missing data for {name}, unable to compute score"
-        measured_mean = measured_mean[0]
-        if fake_data is not None and name in fake_data:
-            # used for sanity checks on the sensitivity of the score metric
-            measured_mean = fake_data[name]
-        benchmark_score = weight * math.log(norm / measured_mean)
-        # print(f"{name}: {benchmark_score}")
-        score += benchmark_score
-
-    score = target * math.exp(score)
-    assert abs(weight_sum - 1.0) < 1e-6, f"Bad configuration, weights don't sum to 1, but {weight_sum}"
-    return score
+def _get_model_task(model_name):
+    """
+    Helper function which extracts the task the model belongs to
+    by iterating over the Model attributes.
+    """
+    MODEL_PATH="../../models"
+    p = Path(__file__).parent.joinpath(MODEL_PATH + model_name)
+    if(p):
+        module = importlib.import_module(f'torchbenchmark.models.{model_name}', package=__name__)
+        Model = getattr(module, 'Model')
+    return Model.task.value
 
+class TorchBenchScore:
+    def __init__(self, spec=SPEC_FILE_DEFAULT, target=TARGET_SCORE_DEFAULT, ref_data=None):
+        self.spec = spec
+        self.target = target
+        self.ref_data = ref_data
+        self.weights = None
+        self.norm = None
+
+    def setup_weights(self):
```

Comment:
i think these 2 functions are good as-is, although i would suggest 2 things
- put _ in the name if it's an internal helper function not intended to be called
- call both from __init__ unless there's a good reason to not have it be ready to use once init'ed?

Reply:
Agreed and fixed now.